Issue 362 options dropdown refactor

shannonlal opened the pull request:
This is an initial refactoring of the options dialog.  I have broken the options dialog into 3 different components.  I have not done any refactoring/optimizing of the code at this point.  In the next PR I will start adding Jest test for the different components.  I am trying to keep the PR's small so I did not add any additional tests at this point.  

@n-riesco If you have few minutes would you mind doing a quick review of this code.  I welcome any comments or suggestions.  

Thanks

n-riesco commented:
I'll review this PR later tonight. But if you have the chance before, could change the PR, so that it doesn't touch `yarn.lock`, please? I know `yarn.lock` is corrupted, but I'm fixing that in #385 (where I'm removing some dependencies).

n-riesco commented on app/components/Settings/OptionsDropdown/OptionsDropdown.react.js:
```diff
@@ -1,61 +1,46 @@
-    }
 
     static propTypes = {
         // See type of react-select's Select `value` property
```
I expected `selectedTable` and  selectedIndex` to be `string` (since we used as keys).
Would `PropTypes.string` break anything?

n-riesco commented on app/components/Settings/OptionsDropdown/OptionsDropdown.react.js:
```diff
@@ -1,61 +1,46 @@
         // See type of react-select's Select `value` property
         selectedTable: PropTypes.any,
         selectedIndex: PropTypes.any,
-
```
I think this line was here to make clear the above comment only applies to `selectedTable` and `selectedIndex`.

n-riesco commented on app/components/Settings/OptionsDropdown/OptionsDropdown.react.js:
```diff
@@ -1,61 +1,46 @@
+        this.renderElasticsearchDocs = this.renderElasticsearchDocs.bind(this);
+    }
+
     renderSQLOptions() {
```
Since this code has moved to `<SQLOptions>`, let's get rid of this method.

n-riesco commented on app/components/Settings/OptionsDropdown/OptionsDropdown.react.js:
```diff
@@ -1,61 +1,46 @@
+        />);
     }
 
     renderElasticsearchIndecies() {
```
Since this code has moved to `<ESIndicesOptions>`, let's get rid of this method.

n-riesco commented on app/components/Settings/OptionsDropdown/OptionsDropdown.react.js:
```diff
@@ -64,38 +49,11 @@ export default class OptionsDropdown extends Component {
+        />);
     }
 
     renderElasticsearchDocs() {
```
Since this code has moved to `<ESDocsOptions>`, let's get rid of this method.

n-riesco commented on app/components/Settings/OptionsDropdown/ESDocsOptions.react.js:
```diff
@@ -0,0 +1,58 @@
+
+export default class ESDocsOptions extends Component {
+    static propTypes = {
+        selectedTable: PropTypes.any,
```
If we can't use `PropTypes.string` here, then we should bring back the comment `// See type of react-select's Select 'value' property`.

n-riesco commented on app/components/Settings/OptionsDropdown/ESIndicesOptions.react.js:
```diff
@@ -0,0 +1,63 @@
+    static propTypes = {
+        elasticsearchMappingsRequest: PropTypes.object,
+        setIndex: PropTypes.func,
+        selectedIndex: PropTypes.any
```
If we can't use `PropTypes.string` here, then we should bring back the comment `// See type of react-select's Select 'value' property`.

n-riesco commented on app/components/Settings/OptionsDropdown/SQLOptions.react.js:
```diff
@@ -0,0 +1,60 @@
+export default class SQLOptions extends Component {
+
+    static propTypes = {
+        selectedTable: PropTypes.any,
```
If we can't use `PropTypes.string` here, then we should bring back the comment `// See type of react-select's Select 'value' property`.

n-riesco commented:
I'd like new files to be lowercase. How about `sql-options.jsx`, `es-indices-options.jsx` and `es-docs-options.jsx`?

n-riesco commented:
Since #385 fixes `yarn.lock` and updates `yarn lint` so that it also checks `.jsx` files, I'd wait for #385 to be merged.

shannonlal commented:
@n-riesco Please ignore the latest commits.  I accidentally pushed to my repo and triggered the build.  I will wait until you have merged #385 and then will resolve the yarn.lock issue.

shannonlal commented:
@n-riesco I fixed most of your comments and removed the yarn.lock from the commit.  Let me know if you have any additional comments or if you want to just merge this in.

Thanks

n-riesco commented:
:dancer: could you do a `squash and merge`, please?

![peek 2018-03-03 20-38](https://user-images.githubusercontent.com/6199391/36939068-02ff338a-1f23-11e8-957c-b614719477a4.gif)

shannonlal commented:
Thanks
